op-e2e/sgt/sgt_test.go

```diff
@@ -0,0 +1,367 @@
+package sgt
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+	"math/rand"
+	"testing"
+
+	op_e2e "github.com/ethereum-optimism/optimism/op-e2e"
+	"github.com/ethereum-optimism/optimism/op-e2e/e2eutils/wait"
+	"github.com/ethereum-optimism/optimism/op-e2e/faultproofs"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	"github.com/ethereum-optimism/optimism/op-service/testutils"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	seqVault  = predeploys.SequencerFeeVaultAddr
+	baseVault = predeploys.BaseFeeVaultAddr
+	l1Vault   = predeploys.L1FeeVaultAddr
+	dummyAddr = common.Address{0xff, 0xff}
+)
+
+func TestSGTDepositFunctionSuccess(t *testing.T) {
+	op_e2e.InitParallel(t)
+	sys, _ := faultproofs.StartFaultDisputeSystem(t)
+	t.Cleanup(sys.Close)
+	ctx := context.Background()
+
+	sgt := NewSgtHelper(t, ctx, sys)
+	depositSgtValue := big.NewInt(10000)
+	_, _, _ = setUpTestAccount(t, ctx, 0, sgt, depositSgtValue, big.NewInt(0))
+}
+
+// Diverse test scenarios to verify that the SoulGasToken(sgt) is utilized for gas payment firstly,
+// unless there is insufficient sgt balance, in which case the native balance will be used instead.
+func TestSGTAsGasPayment(t *testing.T) {
+	op_e2e.InitParallel(t)
+	sys, _ := faultproofs.StartFaultDisputeSystem(t)
```

Comment:
In theory each action test should be independent.
Currently actions share the same `sys`, will it be too slow if actions create their own `sys`?

Reply:
right, it'll be slower